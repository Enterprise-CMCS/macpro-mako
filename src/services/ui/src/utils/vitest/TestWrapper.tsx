import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import { ReactLocation, Router } from "@tanstack/react-location";

const queryClient = new QueryClient();
const location = new ReactLocation();

export const TestWrapper = ({ children }: any) => (
  <QueryClientProvider client={queryClient}>
    <Router location={location} routes={[]}>
      {children}
    </Router>
  </QueryClientProvider>
);
