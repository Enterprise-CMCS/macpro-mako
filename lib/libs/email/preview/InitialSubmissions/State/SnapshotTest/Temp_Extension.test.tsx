import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import TempExtStatePreview from "../Temp_Extension";
describe("Initial Submission State Email Snapshot Test", () => {
  it("renders a TempExt Preview Template", () => {
    const { asFragment } = render(<TempExtStatePreview />);

    expect(asFragment()).toMatchSnapshot();
  });
});
