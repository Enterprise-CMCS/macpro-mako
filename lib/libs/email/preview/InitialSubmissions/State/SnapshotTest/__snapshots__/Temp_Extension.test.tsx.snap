// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Initial Submission State Email Snapshot Test > renders a TempExt Preview Template 1`] = `
<DocumentFragment>
  <meta
    content="text/html; charset=UTF-8"
    http-equiv="Content-Type"
  />
  <meta
    name="x-apple-disable-message-reformatting"
  />
  <div
    style="display: none; overflow: hidden; line-height: 1px; opacity: 0; max-height: 0; max-width: 0;"
  >
    Temporary Extension MD-2343.R00.TE09 Submitted
    <div>
       ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿ ‌​‍‎‏﻿
    </div>
  </div>
  <table
    align="center"
    border="0"
    cellpadding="0"
    cellspacing="0"
    role="presentation"
    style="max-width: 37.5em; background-color: rgb(245, 245, 245);"
    width="100%"
  >
    <tbody>
      <tr
        style="width: 100%;"
      >
        <td>
          <table
            align="center"
            border="0"
            cellpadding="0"
            cellspacing="0"
            role="presentation"
            style="background-color: rgb(0, 113, 189); padding: 16px;"
            width="100%"
          >
            <tbody>
              <tr>
                <td>
                  <a
                    href="https://mako-dev.cms.gov/"
                    style="color: rgb(6, 125, 247); text-decoration: none; display: block; max-width: 112px;"
                    target="_blank"
                  >
                    <img
                      alt="OneMAC Logo"
                      height="40"
                      src="https://mako-dev.cms.gov/onemac-logo.png"
                      style="max-width: 112px;"
                      width="112"
                    />
                  </a>
                </td>
              </tr>
            </tbody>
          </table>
          <div
            style="margin: 0px 8px 8px 8px; padding: 8px;"
          >
            <h1
              style="color: rgb(51, 51, 51); font-size: 20px; font-weight: bold; margin-bottom: 15px;"
            >
              This response confirms you have submitted a Temporary Extension to CMS for review
            </h1>
            <table
              align="center"
              border="0"
              cellpadding="0"
              cellspacing="0"
              role="presentation"
              width="100%"
            >
              <tbody>
                <tr>
                  <td>
                    <table
                      align="center"
                      border="0"
                      cellpadding="0"
                      cellspacing="0"
                      role="presentation"
                      width="100%"
                    >
                      <tbody
                        style="width: 100%;"
                      >
                        <tr
                          style="width: 100%;"
                        >
                          <td
                            align="left"
                            data-id="__react-email-column"
                            style="width: 50%;"
                          >
                            <p
                              style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; font-weight: 600; letter-spacing: -0.5px;"
                            >
                              State or Territory:
                            </p>
                          </td>
                          <td
                            data-id="__react-email-column"
                          >
                            <p
                              style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; color: rgb(51, 51, 51);"
                            >
                              MD
                            </p>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                    <table
                      align="center"
                      border="0"
                      cellpadding="0"
                      cellspacing="0"
                      role="presentation"
                      width="100%"
                    >
                      <tbody
                        style="width: 100%;"
                      >
                        <tr
                          style="width: 100%;"
                        >
                          <td
                            align="left"
                            data-id="__react-email-column"
                            style="width: 50%;"
                          >
                            <p
                              style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; font-weight: 600; letter-spacing: -0.5px;"
                            >
                              Name:
                            </p>
                          </td>
                          <td
                            data-id="__react-email-column"
                          >
                            <p
                              style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; color: rgb(51, 51, 51);"
                            >
                              George Harrison
                            </p>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                    <table
                      align="center"
                      border="0"
                      cellpadding="0"
                      cellspacing="0"
                      role="presentation"
                      width="100%"
                    >
                      <tbody
                        style="width: 100%;"
                      >
                        <tr
                          style="width: 100%;"
                        >
                          <td
                            align="left"
                            data-id="__react-email-column"
                            style="width: 50%;"
                          >
                            <p
                              style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; font-weight: 600; letter-spacing: -0.5px;"
                            >
                              Email Address:
                            </p>
                          </td>
                          <td
                            data-id="__react-email-column"
                          >
                            <p
                              style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; color: rgb(51, 51, 51);"
                            >
                              george@example.com
                            </p>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                    <table
                      align="center"
                      border="0"
                      cellpadding="0"
                      cellspacing="0"
                      role="presentation"
                      width="100%"
                    >
                      <tbody
                        style="width: 100%;"
                      >
                        <tr
                          style="width: 100%;"
                        >
                          <td
                            align="left"
                            data-id="__react-email-column"
                            style="width: 50%;"
                          >
                            <p
                              style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; font-weight: 600; letter-spacing: -0.5px;"
                            >
                              Temporary Extension Request Number:
                            </p>
                          </td>
                          <td
                            data-id="__react-email-column"
                          >
                            <p
                              style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; color: rgb(51, 51, 51);"
                            >
                              MD-2343.R00.TE09
                            </p>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                    <table
                      align="center"
                      border="0"
                      cellpadding="0"
                      cellspacing="0"
                      role="presentation"
                      width="100%"
                    >
                      <tbody
                        style="width: 100%;"
                      >
                        <tr
                          style="width: 100%;"
                        >
                          <td
                            align="left"
                            data-id="__react-email-column"
                            style="width: 50%;"
                          >
                            <p
                              style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; font-weight: 600; letter-spacing: -0.5px;"
                            >
                              Temporary Extension Type:
                            </p>
                          </td>
                          <td
                            data-id="__react-email-column"
                          >
                            <p
                              style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; color: rgb(51, 51, 51);"
                            >
                              1915(b)
                            </p>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                    <table
                      align="center"
                      border="0"
                      cellpadding="0"
                      cellspacing="0"
                      role="presentation"
                      width="100%"
                    >
                      <tbody
                        style="width: 100%;"
                      >
                        <tr
                          style="width: 100%;"
                        >
                          <td
                            align="left"
                            data-id="__react-email-column"
                            style="width: 50%;"
                          >
                            <p
                              style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; font-weight: 600; letter-spacing: -0.5px;"
                            >
                              90th Day Deadline:
                            </p>
                          </td>
                          <td
                            data-id="__react-email-column"
                          >
                            <p
                              style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; color: rgb(51, 51, 51);"
                            >
                              May 31, 2023 @ 11:59pm EST
                            </p>
                          </td>
                        </tr>
                      </tbody>
                    </table>
                    <hr
                      style="width: 100%; border-top: 1px solid #0071BD; margin: 16px 0px;"
                    />
                    <p
                      style="font-size: 14px; line-height: 24px; margin: .5em;"
                    />
                    <h2
                      style="font-size: 16px;"
                    >
                      Summary:
                    </h2>
                    <p />
                    <p
                      style="font-size: 14px; line-height: 1.4; margin: 12px 0px; color: rgb(51, 51, 51);"
                    >
                      Whoever fights monsters should see to it that in the process he does not become a monster. And if you gaze long enough into an abyss, the abyss will gaze back into you.
                    </p>
                    <table
                      align="center"
                      border="0"
                      cellpadding="0"
                      cellspacing="0"
                      role="presentation"
                      width="100%"
                    >
                      <tbody
                        style="width: 100%;"
                      >
                        <tr
                          style="width: 100%;"
                        />
                      </tbody>
                    </table>
                  </td>
                </tr>
              </tbody>
            </table>
            <hr
              style="width: 100%; border-top: 1px solid #0071BD; margin: 16px 0px;"
            />
            <h2
              style="font-size: 16px;"
            >
              Files:
            </h2>
            <table
              align="center"
              border="0"
              cellpadding="0"
              cellspacing="0"
              role="presentation"
              style="margin-bottom: 2px; margin-top: 2px;"
              width="100%"
            >
              <tbody
                style="width: 100%;"
              >
                <tr
                  style="width: 100%;"
                >
                  <td
                    align="left"
                    data-id="__react-email-column"
                    style="width: 50%; vertical-align: top;"
                  >
                     
                    <span>
                      <p
                        style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; font-weight: 600; letter-spacing: -0.5px;"
                      >
                        Waiver Extension Request:
                      </p>
                       
                    </span>
                  </td>
                  <td
                    data-id="__react-email-column"
                    style="vertical-align: top;"
                  >
                    <p
                      style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; color: rgb(51, 51, 51);"
                    >
                      <span>
                        Temporary Extention Document for submission.pdf
                      </span>
                    </p>
                  </td>
                </tr>
              </tbody>
            </table>
            <table
              align="center"
              border="0"
              cellpadding="0"
              cellspacing="0"
              role="presentation"
              style="margin-bottom: 2px; margin-top: 2px;"
              width="100%"
            >
              <tbody
                style="width: 100%;"
              >
                <tr
                  style="width: 100%;"
                >
                  <td
                    align="left"
                    data-id="__react-email-column"
                    style="width: 50%; vertical-align: top;"
                  >
                     
                    <span>
                      <p
                        style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; font-weight: 600; letter-spacing: -0.5px;"
                      >
                        Waiver Extension Request:
                      </p>
                       
                    </span>
                  </td>
                  <td
                    data-id="__react-email-column"
                    style="vertical-align: top;"
                  >
                    <p
                      style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; color: rgb(51, 51, 51);"
                    >
                      <span>
                        Second Extention Document for submission.pdf
                      </span>
                    </p>
                  </td>
                </tr>
              </tbody>
            </table>
            <table
              align="center"
              border="0"
              cellpadding="0"
              cellspacing="0"
              role="presentation"
              style="margin-bottom: 2px; margin-top: 2px;"
              width="100%"
            >
              <tbody
                style="width: 100%;"
              >
                <tr
                  style="width: 100%;"
                >
                  <td
                    align="left"
                    data-id="__react-email-column"
                    style="width: 50%; vertical-align: top;"
                  >
                     
                    <span>
                      <p
                        style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; font-weight: 600; letter-spacing: -0.5px;"
                      >
                        Waiver Extension Request:
                      </p>
                       
                    </span>
                  </td>
                  <td
                    data-id="__react-email-column"
                    style="vertical-align: top;"
                  >
                    <p
                      style="font-size: 14px; line-height: 1.4; margin: 4px 0px 4px 0px; color: rgb(51, 51, 51);"
                    >
                      <span>
                        Third Extention Document for submission.pdf
                      </span>
                    </p>
                  </td>
                </tr>
              </tbody>
            </table>
            <p
              style="font-size: 14px; line-height: 1.4; margin: 16px 0px 16px 0px; color: rgb(51, 51, 51);"
            >
              This mailbox is for the submittal of Section 1915(b) and 1915(c) Waivers, responses to Requests for Additional Information (RAI) on Waivers, and extension requests on Waivers only. Any other correspondence will be disregarded.
            </p>
            <p
              style="font-size: 14px; line-height: 1.4; margin: 12px 0px; color: rgb(51, 51, 51);"
            >
              Thank you.
            </p>
          </div>
          <table
            align="center"
            border="0"
            cellpadding="0"
            cellspacing="0"
            role="presentation"
            style="font-size: 14px; padding: 0px 0px; font-weight: 300; background-color: rgb(0, 113, 189); color: rgb(255, 255, 255); text-align: center;"
            width="100%"
          >
            <tbody>
              <tr>
                <td>
                  <hr
                    style="width: 100%; border-top: 1px solid #0071BD; margin: 16px 0px;"
                  />
                  <table
                    align="center"
                    border="0"
                    cellpadding="0"
                    cellspacing="0"
                    role="presentation"
                    width="100%"
                  >
                    <tbody>
                      <tr>
                        <td>
                          <p
                            style="font-size: 14px; line-height: 24px; margin: 8px 0px 16px 0px;"
                          >
                            If you have any questions, please contact 
                            <a
                              href="mailto:spa@cms.hhs.gov"
                              style="color: rgb(6, 125, 247); text-decoration: underline;"
                              target="_blank"
                            >
                              spa@cms.hhs.gov
                            </a>
                             or your state lead.
                          </p>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </td>
              </tr>
            </tbody>
          </table>
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
